flixel/tile/FlxBaseTilemap.hx

```diff
@@ -730,19 +733,70 @@ class FlxBaseTilemap<Tile:FlxObject> extends FlxObject
 		{
 			return null;
 		}
-		
+
+		var zone:FlxTilemapZone = switch (PathingMethod)
+		{
+			// Basic pathing method where distances from starting point is calculated for all tiles
+			// until reaching the end point, and then path is found by backtracking from end point to starting point
+			case FlxTilemapPathingMethod.BASIC:
+				{x0:0, y0:0, sizeX:widthInTiles, sizeY:heightInTiles, indexStart:startIndex, indexEnd:endIndex};
+
+			// Scaled horizon method should be used with large maps:
+			// the basic pathing method is used over a small zone
+			case FlxTilemapPathingMethod.BASIC_SCALED_HORIZON(scale, minHorizonSize):
+				if (scale < 1.0) scale = 1.0;
+				var startX = (startIndex % widthInTiles);
+				var startY = Math.floor(startIndex / widthInTiles);
+				var endX = (endIndex % widthInTiles);
+				var endY = Math.floor(endIndex / widthInTiles);
+
+				var leftX = if (startX < endX) startX else endX;
+				var rightX = if (startX < endX) endX else startX;
+				var highY = if (startY < endY) startY else endY;
+				var lowY = if (startY < endY) endY else startY;
+
+				var deltaX = rightX - leftX;
+				if (deltaX < minHorizonSize) deltaX = minHorizonSize;
+				var deltaY = lowY - highY;
+				if (deltaY < minHorizonSize) deltaY = minHorizonSize;
+
+				var x0 = Math.floor(leftX - scale * 0.5 * deltaX);
+				if (x0 < 0) x0 = 0;
+				var x1 = Math.ceil(rightX + scale * 0.5 * deltaX);
+				if (x1 > widthInTiles) x1 = widthInTiles;
+				var y0 = Math.floor(highY - scale * 0.5 * deltaY);
+				if (y0 < 0) y0 = 0;
+				var y1 = Math.ceil(lowY + scale * 0.5 * deltaY);
+				if (y1 > heightInTiles) y1 = heightInTiles;
+
+				var zoneSizeX = x1 - x0;
+				var zoneSizeY = y1 - y0;
+				var startIndexInZone = (startX - x0) + (startY - y0) * zoneSizeX;
+				var endIndexInZone = (endX - x0) + (endY - y0) * zoneSizeX;
+				{x0:x0, y0:y0, sizeX:zoneSizeX, sizeY:zoneSizeY, indexStart:startIndexInZone, indexEnd:endIndexInZone};
```

Comment:
Same problem here, this code likely generates Dynamic objects. We should create a classes.